bst.py

```diff
@@ -113,6 +111,48 @@ def _get_dot(self, current):
             yield "\tnull%s [shape=point];" % r
             yield "\t%s -> null%s;" % (current, r)
 
+    def in_order(self, current='start'):
+        """
+        Generator that traverses the binary tree.
+        """
+        if current == 'start':
+            current = self.top
+        if current is not None:
+            self.in_order(self.tree[current].get('left')).next()
+            yield current
+            self.in_order(self.tree[current].get('right')).next()
```

Comment:
These functions do not work the way you think they do.  Have you tried to play with them at the command line?

```python
>>> from bst import Bst
>>> tree = Bst()
>>> for val in [5,3,7,4,1,6,8]:
...   tree.insert(val)
...
>>> tree.top
5
>>> tree.tree
{1: {'depth': 3}, 3: {'depth': 2, 'right': 4, 'left': 1}, 4: {'depth': 3}, 5: {'depth': 1, 'right': 7, 'left': 3}, 6: {'depth': 3}, 7: {'depth': 2, 'right': 8, 'left': 6}, 8: {'depth': 3}}
>>> [val for val in tree.in_order()]
[]
>>> [val for val in tree.pre_order()]
[5]
>>> [val for val in tree.post_order()]
[]
>>> [val for val in tree.breadth_first()]
[5, 3, 7, 1, 4, 6, 8]
```